Production/src/SVfitInterface.cc

```diff
@@ -2,25 +2,32 @@
 This file is part of https://github.com/hh-italian-group/h-tautau. */
 
 #include "../interface/SVfitInterface.h"
-#include "TauAnalysis/SVfitStandalone/interface/SVfitStandaloneAlgorithm.h"
+#include "TauAnalysis/ClassicSVfit/interface/ClassicSVfit.h"
+#include "TauAnalysis/ClassicSVfit/interface/MeasuredTauLepton.h"
+#include "TauAnalysis/ClassicSVfit/interface/svFitHistogramAdapter.h"
 
 namespace analysis {
 
 namespace sv_fit {
 
-FitResults FitProducer::RunAlgorithm(const std::vector<svFitStandalone::MeasuredTauLepton>& measured_leptons,
+FitResults FitProducer::RunAlgorithm(const std::vector<classic_svFit::MeasuredTauLepton>& measured_leptons,
                                      const LorentzVector& met_momentum, const SquareMatrix<2>& met_cov) const
 {
     const TMatrixD met_cov_t = ConvertMatrix(met_cov);
-    SVfitStandaloneAlgorithm algo(measured_leptons, met_momentum.Px(), met_momentum.Py(), met_cov_t, verbosity);
-    algo.addLogM(false);
-    algo.shiftVisPt(true, visPtResolutionFile.get());
-    algo.integrateMarkovChain();
+    ClassicSVfit algo(verbosity);
+    algo.addLogM_fixed(false);
+    // algo.setDiTauMassConstraint(-1.0);
+    algo.integrate(measured_leptons, met_momentum.Px(), met_momentum.Py(), met_cov_t);
 
     FitResults result;
     if(algo.isValidSolution()) {
-        result.momentum = LorentzVectorM(algo.pt(), algo.eta(), algo.phi(), algo.mass());
-        result.transverseMass = algo.transverseMass();
+        double pt = static_cast<classic_svFit::DiTauSystemHistogramAdapter*>(algo.getHistogramAdapter())->getPt();
+        double eta = static_cast<classic_svFit::DiTauSystemHistogramAdapter*>(algo.getHistogramAdapter())->getEta();
+        double phi = static_cast<classic_svFit::DiTauSystemHistogramAdapter*>(algo.getHistogramAdapter())->getPhi();
+        double mass = static_cast<classic_svFit::DiTauSystemHistogramAdapter*>(algo.getHistogramAdapter())->getMass();
```

Comment:
create auto histAdapter and call this method only once.
Better to use dynamic_cast.